weave-js/src/context.tsx

```diff
@@ -13,11 +13,14 @@ export type WeaveWBBetaFeatures = {
   'weave-devpopup'?: boolean;
 };
 
+export type PanelSettingPanel = 'JupyterViewer';
+
 export interface WeaveFeatures {
   actions?: boolean;
   fullscreenMode?: boolean;
   dashUi?: boolean;
   betaFeatures: WeaveWBBetaFeatures;
+  panelSettings?: Record<PanelSettingPanel, any>;
```

Comment:
This usage of WeaveFeatures is correct per @nickpenaranda (and vetted with  shawn)

Reply:
yeah, I took a quick pass at that, and it would have ended up being a union type like `{JupyterView: MyJupyterType}|{otherPanel: OtherType}`, and I wasn't clear on how to say you could pass a map of those cleanly, and trying to define things as `Record<PanelSettingPanel, x>` I knew there wouldn't be a way to restrict whatever type x is to something meaningful.

I suspect there must be a way to define that kind of type though.If you want to push a change that works with the current code, I'd be happy with that.